src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def navigate_to_obj_pos(  # type: ignore
+        env,
+        obj,
+        pos_offset: Array,
+        rng: Generator = np.random.default_rng(23),
+) -> Optional[Callable]:
+    """Parameterized controller for navigation.
+
+    Runs motion planning to find a feasible trajectory to a certain x,y
+    position offset from obj and selects an orientation such that the
+    robot is facing the object. If the navigation is infeasible, returns
+    an indication to this effect (None). Otherwise, returns a function
+    that can be stepped like an option to output actions at each
+    timestep.
+    """
+    # test agent positions around an obj
+    # try to place the agent near the object, and rotate it to the object
+    valid_position = None  # ((x,y,z),(roll, pitch, yaw))
+    original_orientation = env.robots[0].get_orientation()
+
+    state = p.saveState()
+
+    def sample_fn(env: BehaviorEnv,
+                  rng: Generator) -> Tuple[float, float, float]:
+        random_point = env.scene.get_random_point(rng=rng)
+        x, y = random_point[1][:2]
+        theta = (
+            rng.random() *
+            (CIRCULAR_LIMITS[1] - CIRCULAR_LIMITS[0])) + CIRCULAR_LIMITS[0]
+        return (x, y, theta)
+
+    if not isinstance(
+            obj,
+            URDFObject):  # must be a URDFObject so we can get its position!
+        print("ERROR! Object to navigate to is not valid (not an instance of" +
+              "URDFObject).")
+        p.restoreState(state)
+        p.removeState(state)
+        print(
+            f"PRIMITIVE: navigate to {obj.name} with params {pos_offset} fail")
+        return None
+
+    obj_pos = obj.get_position()
+    pos = [
+        pos_offset[0] + obj_pos[0],
+        pos_offset[1] + obj_pos[1],
+        env.robots[0].initial_z_offset,
+    ]
+    yaw_angle = np.arctan2(pos_offset[1], pos_offset[0]) - np.pi
+    orn = [0, 0, yaw_angle]
+    env.robots[0].set_position_orientation(pos, p.getQuaternionFromEuler(orn))
+    eye_pos = env.robots[0].parts["eye"].get_position()
+    ray_test_res = p.rayTest(eye_pos, obj_pos)
+    # Test to see if the robot is obstructed by some object, but make sure
+    # that object is not either the robot's body or the object we want to
+    # pick up!
+    blocked = len(ray_test_res) > 0 and (ray_test_res[0][0] not in (
+        env.robots[0].parts["body"].get_body_id(),
+        obj.get_body_id(),
+    ))
+    if not detect_robot_collision(env.robots[0]) and not blocked:
+        valid_position = (pos, orn)
+
+    if valid_position is None:
+        print("Position commanded is in collision or blocked!")
+        p.restoreState(state)
+        p.removeState(state)
+        print(
+            f"PRIMITIVE: navigate to {obj.name} with params {pos_offset} fail")
+        return None
+
+    p.restoreState(state)
+    p.removeState(state)
+    state = p.saveState()
```

Comment:
what's going on in these 3 lines? why remove the state and then save it again?

Reply:
that's a relic of old code, thanks for pointing out!